src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -1872,6 +1872,19 @@ ReturnCode_t DataWriterImpl::check_qos(
         EPROSIMA_LOG_ERROR(RTPS_QOS_CHECK, "DATA_SHARING cannot be used with memory policies other than PREALLOCATED.");
         return ReturnCode_t::RETCODE_INCONSISTENT_POLICY;
     }
+    if (qos.history().kind == KEEP_LAST_HISTORY_QOS && qos.history().depth <= 0)
```

Comment:
At the end I dont know if we should let `KEEP_LAST` with a `depth` of `-1` . What do you think @MiguelCompany ?

Reply:
We should **NOT** let `KEEP_LAST` with a `depth` of `-1`, so that if-else expression should not be updated